Vacuum tool throws ClassNotFoundException

When running version 2.0.0 the following exception is thrown:

> com.hotels.housekeeping.HousekeepingConfiguration.housekeepingEnvironment(HousekeepingConfiguration.java:68)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f.CGLIB$housekeepingEnvironment$0(<generated>)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f$$FastClassBySpringCGLIB$$a56e3fa4.invoke(<generated>)
>     at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
>     at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:355)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f.housekeepingEnvironment(<generated>)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:498)
>     at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
>     ... 70 more
> Caused by: java.lang.ClassNotFoundException: com.hotels.shaded.com.google.common.collect.ImmutableMap
>     at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
>     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
>     ... 81 more

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,9 @@
 ## [TBD] - TBD
 ### Changed
 * Refactored general metastore tunnelling code to leverage hcommon-hive-metastore libraries. See [#23](https://github.com/HotelsDotCom/housekeeping/issues/23).
+* Upgraded hotels-oss-parent to 2.3.3 (was 2.0.6).
+* Packaging vaccuum-tool results in a jar with a new name: housekeeping-vacuum-tool-all-latest.jar.
+* Removed the hcommon-hive-metastore jar from the lib folder and shaded it into housekeeping-vacuum-tool.
 
 ## [2.0.0] - 2018-06-22
 ### Changed

--- a/housekeeping-vacuum-tool/README.md
+++ b/housekeeping-vacuum-tool/README.md
@@ -29,7 +29,12 @@
 
 ##### Vacuum Tool configured with MySQL Housekeeping database
 
-Add your JDBC driver to `$VACUUM_TOOL_HOME/lib/` or some other location on the classpath of the Vacuum tool.
+In order to use an external JDBC-compliant database, the JDBC driver for this database must be made available on the CLASSPATH of the vacuum tool. 
+This can be achieved by one of the following:
+* Adding the path to the driver jar file to the Housekeeping bootstrap CLASSPATH (e.g. `export HOUSEKEEPING_CLASSPATH=/path/to/mysql-connector-java-x.y.z.jar`). 
+* Placing the driver jar file in `$VACUUM_TOOL_HOME/lib/`.
+
+The configuration then needs to be updated to be something like below:
 
     catalog:
       name: vacuum_tool
@@ -52,9 +57,9 @@ Add your JDBC driver to `$VACUUM_TOOL_HOME/lib/` or some other location on the c
         password: foo
 
 
-
-
 ##### Vacuum Tool configured with H2 Housekeeping database
+
+The Vacuum tool already has the required H2 drivers on its CLASSPATH so the only change required to use H2 is to create a configuration file similar to below:
 
     catalog:
       name: vacuum_tool

--- a/housekeeping-vacuum-tool/pom.xml
+++ b/housekeeping-vacuum-tool/pom.xml
@@ -82,7 +82,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.3</version>
+        <version>3.2.0</version>
         <executions>
           <execution>
             <phase>package</phase>
@@ -95,6 +95,7 @@
               <artifactSet>
                 <includes>
                   <include>com.hotels:housekeeping-vacuum-tool</include>
+                  <include>com.hotels:hcommon-hive-metastore</include>
                   <include>com.google.guava:guava</include>
                 </includes>
               </artifactSet>

--- a/housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml
+++ b/housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml
@@ -19,6 +19,7 @@
       <useProjectArtifact>false</useProjectArtifact>
       <excludes>
         <exclude>com.google.guava:guava</exclude>
+        <exclude>com.hotels:hcommon-hive-metastore</exclude>
       </excludes>
     </dependencySet>
   </dependencySets>

--- a/housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml
+++ b/housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml
@@ -1,14 +1,25 @@
-<assembly xmlns="http://maven.apache.org/ASSEMBLY/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-          xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.0.0 http://maven.apache.org/xsd/assembly-2.0.0.xsd">
+<assembly xmlns="http://maven.apache.org/ASSEMBLY/2.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/ASSEMBLY/2.0.0 http://maven.apache.org/xsd/assembly-2.0.0.xsd">
   <id>bin</id>
   <formats>
     <format>tgz</format>
   </formats>
   <baseDirectory>housekeeping-vacuum-tool-${project.version}</baseDirectory>
+  <files>
+    <file>
+      <source>target/housekeeping-vacuum-tool-${project.version}-all.jar</source>
+      <outputDirectory>lib</outputDirectory>
+      <destName>housekeeping-vacuum-tool-all-latest.jar</destName>
+    </file>
+  </files>
   <dependencySets>
     <dependencySet>
       <outputDirectory>lib</outputDirectory>
-      <useProjectArtifact>true</useProjectArtifact>
+      <useProjectArtifact>false</useProjectArtifact>
+      <excludes>
+        <exclude>com.google.guava:guava</exclude>
+      </excludes>
     </dependencySet>
   </dependencySets>
   <fileSets>

--- a/pom.xml
+++ b/pom.xml
@@ -5,9 +5,9 @@
   <parent>
     <groupId>com.hotels</groupId>
     <artifactId>hotels-oss-parent</artifactId>
-    <version>2.0.6</version>
+    <version>2.3.3</version>
   </parent>
-
+  
   <groupId>com.hotels</groupId>
   <artifactId>housekeeping-parent</artifactId>
   <version>2.1.0-SNAPSHOT</version>
@@ -31,8 +31,6 @@
     <hadoop.version>2.7.1</hadoop.version>
     <hive.version>2.3.2</hive.version>
     <spring-platform.version>2.0.3.RELEASE</spring-platform.version>
-    <junit.version>4.12</junit.version>
-    <mockito.version>1.10.19</mockito.version>
     <maven-assembly-plugin.version>3.1.0</maven-assembly-plugin.version>
   </properties>
 
@@ -70,13 +68,11 @@
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>${junit.version}</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
-      <version>${mockito.version}</version>
       <scope>test</scope>
     </dependency>
   </dependencies>

